ironfish/src/rpc/routes/wallet/rescan.ts

```diff
@@ -0,0 +1,64 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+import * as yup from 'yup'
+import { ApiNamespace } from '../namespaces'
+import { routes } from '../router'
+import { AssertHasRpcContext } from '../rpcContext'
+
+export type RescanRequest = {
+  follow?: boolean
+  from?: number
+}
+export type RescanResponse = { sequence: number; startedAt: number; endSequence: number }
+
+export const RescanRequestSchema: yup.ObjectSchema<RescanRequest> = yup
+  .object({
+    follow: yup.boolean().optional(),
+    from: yup.number().optional(),
+    full: yup.boolean().optional(),
+  })
+  .defined()
+
+export const RescanResponseSchema: yup.ObjectSchema<RescanResponse> = yup
+  .object({
+    sequence: yup.number().defined(),
+    endSequence: yup.number().defined(),
+    startedAt: yup.number().defined(),
+  })
+  .defined()
+
+routes.register<typeof RescanRequestSchema, RescanResponse>(
+  `${ApiNamespace.wallet}/rescan`,
```

Comment:
Noting that this is a breaking change renaming `rescanAccount` to `rescan`.

Reply:
Added a note for this.